### PR TITLE
Upgrade USWDS to 3.5.0

### PR DESCRIPTION
**Pull request by namanaman**
- Upgrades USWDS dependency from 3.0.1 to 3.5.0
- Pulls in new components from USWDS
- Updates README on how to release a new package version

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -1,6 +1,6 @@
 MIT License
 
-Copyright (c) 2020 New Jersey
+Copyright (c) 2023 New Jersey
```
Comment (casewalker) [inline LICENSE:3]: Nice catch!

File: README.md [hunk 2]
```diff
@@ -54,3 +54,12 @@ This imports the USWDS, pulls in the NJ-specific components and styles, and save
 This builds USWDS styles, builds the Fractal docs, and then deploys them to the gh-pages branch.
 
 Note: do not push directly to the gh-pages branch.
+
+## Releasing a new version to NPM
+
+1. Go to the Draft Release Github Action https://github.com/newjersey/njwds/actions/workflows/draft-release.yml
```
Comment (casewalker) [inline README.md:60]: Tiny nit: This readme has both "Github" and "GitHub" (capital "H").
It looks like the company prefers "GitHub". Please update that here, on line 64, and back on line 19 as well.

Comment (casewalker) [inline README.md:60]: Also, it looks like the rest of the readme uses "link text" every time, we should probably change all of the links in this section to match as well, e.g:
```
1. Go to the [Draft Release GitHub Action](https://github.com/newjersey/njwds/actions/workflows/draft-release.yml)
...
```

File: README.md [hunk 3]
```diff
@@ -54,3 +54,12 @@ This imports the USWDS, pulls in the NJ-specific components and styles, and save
...
+2. Click Run workflow dropdown, keep the branch set to `main`, and update the Semver Level based on what has changed since the previous release (see here for details: https://semver.org/). Click green "Run workflow" button.
+3. Confirm that this action ran successfully by checking that the `package.json` file's version has been bumped according to the level you set to the release (i.e. minor release changes version from 0.1.0 to 0.2.0). Go to https://github.com/newjersey/njwds/releases and confirm that you see a new draft release with this version.
```
Comment (casewalker) [inline README.md:62]: Nit: I think this is an "e.g." instead of an "i.e."

File: src/components/card/card--flag.njk [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+
+      <div class="usa-card__body">
+        <p>Lorem ipsum dolor sit amet consectetur adipisicing elit.</p>
+      </div>
+      <div class="usa-card__footer">
+        <button class="usa-button">Visit Florida Keys</button>
+      </div>
+    </div>
+  </li>
+</ul>
```
Comment (casewalker) [inline src/components/card/card--flag.njk:40]: Do we control these files at all? Should we? If so, please add a newline at the end of this file.

File: src/components/checkbox/checkbox--test.njk [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+    <div class="usa-checkbox">
+      <input class="usa-checkbox__input usa-checkbox__input--tile" id="check-historical-washington-3" type="checkbox" name="historical-figures-3" value="booker-t-washington">
+      <label class="usa-checkbox__label" for="check-historical-washington-3">Booker T. Washington</label>
+    </div>
+    <div class="usa-checkbox">
+      <input class="usa-checkbox__input usa-checkbox__input--tile" id="check-historical-carver-3" type="checkbox" name="historical-figures-3"  value="george-washington-carver" disabled>
+      <label class="usa-checkbox__label" for="check-historical-carver-3">George Washington Carver</label>
+    </div>
+  </fieldset>
+</div>
```
Comment (casewalker) [inline src/components/checkbox/checkbox--test.njk:41]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/collection/collection--only-headers.njk [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+        <li class="usa-collection__meta-item position-relative">
+          <svg class="usa-icon position-relative bottom-neg-2px" aria-hidden="true" role="img">
+            <use xlink:href="../../dist/img/sprite.svg#public"></use>
+          </svg>
+          Performance.gov
+        </li>
+      </ul>
+    </div>
+  </li>
+</ul>
```
Comment (casewalker) [inline src/components/collection/collection--only-headers.njk:69]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/combo-box/combo-box--test.njk [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    merge=true %}
+  {% render "@combo-box",
+    { comboBox: {
+        utilities: "width-full maxw-none",
+        label: "Not in a usa-form element, custom width utilities",
+        name: "even-moar-moar-fruit"
+      }
+    },
+    merge=true %}
+</div>
```
Comment (casewalker) [inline src/components/combo-box/combo-box--test.njk:34]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/type/links.njk [hunk 8]
```diff
@@ -0,0 +1,10 @@
+<p>This is <a class="usa-link" href="javascript:void(0);">a text link</a> on a light background.</p>
+
+<p>This is <a class="usa-link usa-color-text-visited" href="javascript:void(0);">a visited link</a>.</p>
+
+<p>This is a link that goes to an <a class="usa-link usa-link--external" href="https://i.giphy.com/media/WPzQF6ruiIIVzHNlwX/source.gif">external website</a>.</p>
+
+<div class="usa-dark-background padding-1 display-inline-block">
+  <p>This is <a class="usa-link" href="javascript:void(0);">a text link on a dark background</a>.</p>
+  <p><a class="usa-link usa-link--alt usa-link--external" href="https://i.giphy.com/media/WPzQF6ruiIIVzHNlwX/source.gif">This</a> is an alternate external text link on a dark background.</p>
+</div>
```
Comment (casewalker) [inline src/components/type/links.njk:10]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/tooltip/tooltip.njk [hunk 9]
```diff
@@ -0,0 +1,14 @@
...
+  <div class="margin-4">
+    <button type="button" class="usa-button usa-tooltip" data-position="right" title="Right">Show on right</button>
+  </div>
+  <div class="margin-4">
+    <button type="button" class="usa-button usa-tooltip" data-position="left" title="Left">Show on left</button>
+  </div>
+  <div class="margin-4">
+    <button type="button" class="usa-button usa-tooltip" data-position="bottom" title="Bottom">Show on bottom</button>
+  </div>
+</div>
```
Comment (casewalker) [inline src/components/tooltip/tooltip.njk:14]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/tooltip/tooltip--utilities.njk [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+  <div class="tablet:grid-col-3 margin-top-2 text-center">
+    <button type="button" class="usa-button usa-tooltip" data-position="bottom" data-classes="{{tooltip.classes}}" title="Bottom">Show on bottom</button>
+  </div>
+  <div class="tablet:grid-col-3 margin-top-2 text-center">
+    <button type="button" class="usa-button usa-tooltip" data-position="right" data-classes="{{tooltip.classes}}" title="Right">Show on right</button>
+  </div>
+  <div class="tablet:grid-col-3 margin-top-2 text-center">
+    <button type="button" class="usa-button usa-tooltip" data-position="left" data-classes="{{tooltip.classes}}" title="Left">Show on left</button>
+  </div>
+</div>
```
Comment (casewalker) [inline src/components/tooltip/tooltip--utilities.njk:14]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/table/table.config.yml [hunk 11]
```diff
@@ -0,0 +1,3 @@
+label: Table
+preview: "@uswds-framed"
+status: ready
```
Comment (casewalker) [inline src/components/table/table.config.yml:3]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/table/table--stacked.njk [hunk 12]
```diff
@@ -0,0 +1,113 @@
...
+<style scoped>
+
+.table-example-container {
+  display: flex;
+  column-gap: 2rem;
+  row-gap: 2rem;
+  flex-flow: row wrap;
+  justify-content: space-around;
+}
+</style>
```
Comment (casewalker) [inline src/components/table/table--stacked.njk:113]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/table/table--stacked.config.yml [hunk 13]
```diff
@@ -0,0 +1,3 @@
+label: Responsive Stacked Table
+preview: "@uswds-framed"
+status: ready
```
Comment (casewalker) [inline src/components/table/table--stacked.config.yml:3]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/table/table--sortable.njk [hunk 14]
```diff
@@ -0,0 +1,232 @@
...
+        <td data-sort-value="0" class="font-mono-sm text-tabular text-right">0</td>
+        <td data-sort-value="4">Fourth</td>
+        <td data-sort-value="4">4th</td>
+        <td data-sort-value="1608114345343">December 16, 2020</td>
+      </tr>
+    </tbody>
+  </table>
+  <div class="usa-sr-only usa-table__announcement-region" aria-live="polite"></div>
+</div>
+
```
Comment (casewalker) [inline src/components/table/table--sortable.njk:232]: Does this file actually have two newlines at the end!?! Spread the love around XD

File: src/components/table/table--sortable.config.yml [hunk 15]
```diff
@@ -0,0 +1,3 @@
+label: Sortable Table
+preview: "@uswds-framed"
+status: ready
```
Comment (casewalker) [inline src/components/table/table--sortable.config.yml:3]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/table/table--scrollable.config.yml [hunk 16]
```diff
@@ -0,0 +1,3 @@
+label: Scrollable Table
+preview: "@uswds-framed"
+status: ready
```
Comment (casewalker) [inline src/components/table/table--scrollable.config.yml:3]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/table/table--borderless.config.yml [hunk 17]
```diff
@@ -0,0 +1,3 @@
+label: Borderless Table
+preview: "@uswds-framed"
+status: ready
```
Comment (casewalker) [inline src/components/table/table--borderless.config.yml:3]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/summary-box/summary-box.njk [hunk 18]
```diff
@@ -0,0 +1,22 @@
...
+          and
+          <a class="usa-summary-box__link" href="javascript:void(0);">hypothermia</a>.</li>
+        <li>Gather emergency supplies for your
+          <a class="usa-summary-box__link" href="javascript:void(0);">home</a>
+          and your
+          <a class="usa-summary-box__link" href="javascript:void(0);">car</a>.</li>
+      </ul>
+    </div>
+  </div>
+</div>
```
Comment (casewalker) [inline src/components/summary-box/summary-box.njk:22]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/radio-buttons/radio-buttons.config.yml [hunk 19]
```diff
@@ -0,0 +1 @@
+preview: "@uswds-framed"
```
Comment (casewalker) [inline src/components/radio-buttons/radio-buttons.config.yml:1]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/process-list/process-list--no-text.njk [hunk 20]
```diff
@@ -0,0 +1,11 @@
...
+  <li class="usa-process-list__item padding-bottom-4">
+    <p class="usa-process-list__heading font-sans-xl line-height-sans-1">Start a process.</p>
+  </li>
+  <li class="usa-process-list__item padding-bottom-4">
+    <p class="usa-process-list__heading font-sans-xl line-height-sans-1">Proceed to the second step.</p>
+  </li>
+  <li class="usa-process-list__item">
+    <p class="usa-process-list__heading font-sans-xl line-height-sans-1">Complete the step-by-step process.</p>
+  </li>
+</ol>
```
Comment (casewalker) [inline src/components/process-list/process-list--no-text.njk:11]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/process-list/process-list--custom-sizing.njk [hunk 21]
```diff
@@ -0,0 +1,14 @@
...
+  </li>
+  <li class="usa-process-list__item padding-bottom-4">
+    <h4 class="usa-process-list__heading font-sans-xl line-height-sans-1">Proceed to the second step.</h4>
+    <p class="font-sans-lg margin-top-1 text-light">Suspendisse id velit vitae ligula volutpat condimentum. Aliquam erat volutpat.</p>
+  </li>
+  <li class="usa-process-list__item">
+    <h4 class="usa-process-list__heading font-sans-xl line-height-sans-1">Complete the step-by-step process.</h4>
+    <p class="font-sans-lg margin-top-1 text-light">Lorem ipsum dolor sit amet, consectetuer adipiscing elit. Morbi commodo, ipsum sed pharetra gravida, orci magna rhoncus neque.</p>
+  </li>
+</ol>
```
Comment (casewalker) [inline src/components/process-list/process-list--custom-sizing.njk:14]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/navigation/sidenav/sidenav.config.yml [hunk 22]
```diff
@@ -0,0 +1,73 @@
...
+              - text: Child link
+                links:
+                  - text: Grandchild link
+                  - text: Grandchild link
+                  - text: Grandchild link
+                    current: true
+              - text: Child link
+              - text: Child link
+              - text: Child link
+          - text: Parent link
```
Comment (casewalker) [inline src/components/navigation/sidenav/sidenav.config.yml:73]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/layouts/create-account/create-account.njk [hunk 23]
```diff
@@ -0,0 +1,14 @@
...
+{% endif %}
+
+{% block header %}
+  {% render '@layout--header-empty' %}
+{% endblock %}
+
+{% block body %}
+  {% include 'layouts/create-account/_create-account-inner.njk' %}
+{% endblock %}
+
```
Comment (casewalker) [inline src/components/layouts/create-account/create-account.njk:14]: Extra newline at the end!?! Amazing! Please only one though!

File: src/components/input-prefix-suffix/input-prefix-suffix.njk [hunk 24]
```diff
@@ -0,0 +1,30 @@
...
+  </div>
+
+  <label class="usa-label" for="example-input-suffix">Weight, in pounds</label>
+  <div class="usa-input-group usa-input-group--sm">
+    <input type="text" id="example-input-suffix" class="usa-input" pattern="[0-9]*" inputmode="numeric">
+    <div class="usa-input-suffix" aria-hidden="true">
+      lbs.
+    </div>
+  </div>
+</form>
```
Comment (casewalker) [inline src/components/input-prefix-suffix/input-prefix-suffix.njk:30]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/icon/icon--sizes.njk [hunk 25]
```diff
@@ -0,0 +1,39 @@
...
+          <use xlink:href="../../dist/img/sprite.svg#accessibility_new"></use>
+        </svg>
+      </td>
+      <td><code>{{item.units}}</code></td>
+      <td><code>{{item.pixels}}</code></td>
+    </tr>
+  {% endfor %}
+    </tbody>
+  </table>
+</div>
```
Comment (casewalker) [inline src/components/icon/icon--sizes.njk:39]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/icon/icon.config.yml [hunk 26]
```diff
@@ -0,0 +1,500 @@
...
+      - name: work
+        meta: "briefcase"
+      - name: youtube
+        meta: "social media"
+      - name: zoom_in
+        meta: "expand enlarge magnifying glass"
+      - name: zoom_out_map
+        meta: "expand fullscreen"
+      - name: zoom_out
+        meta: "shrink magnifying glass"
```
Comment (casewalker) [inline src/components/icon/icon.config.yml:500]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/date-range-picker/date-range-picker.config.yml [hunk 27]
```diff
@@ -0,0 +1 @@
+preview: "@uswds-form"
```
Comment (casewalker) [inline src/components/date-range-picker/date-range-picker.config.yml:1]: Please add a newline at the end of the file (if we control these files at all)

File: src/components/date-input/date-input.config.yml [hunk 28]
```diff
@@ -0,0 +1 @@
+preview: "@uswds-form"
```
Comment (casewalker) [inline src/components/date-input/date-input.config.yml:1]: Please add a newline at the end of the file (if we control these files at all)

File: README.md [hunk 29]
```diff
@@ -54,3 +54,12 @@ This imports the USWDS, pulls in the NJ-specific components and styles, and save
 This builds USWDS styles, builds the Fractal docs, and then deploys them to the gh-pages branch.
 
 Note: do not push directly to the gh-pages branch.
+
+## Releasing a new version to NPM
+
+1. Go to the Draft Release Github Action https://github.com/newjersey/njwds/actions/workflows/draft-release.yml
+2. Click Run workflow dropdown, keep the branch set to `main`, and update the Semver Level based on what has changed since the previous release (see here for details: https://semver.org/). Click green "Run workflow" button.
```
Comment (casewalker) [inline README.md:61]: I think maybe this should be `Click the "Run workflow" dropdown` and `Click the green...`

**Review by casewalker — COMMENTED** [review]
Hey hey. Not a lot to review here, looks good and is very exciting, but I left some comments about the Readme text and a lot of comments trying to standardize the ends of the USWDS files. However, maybe it is best that we just don't touch those files at all? Not sure, curious to know what you think.

Thanks!

**Comment by dhcole** [comment]
@namanaman I notice two things here: https://pr-37.d6umhtb6a6pvv.amplifyapp.com/components/preview/header--extended.html

* The search icon appears broken
* The contrast between the utlity links and the white background seems too low

Not sure if these are upstream issues or something we introduced

**Comment by dhcole** [comment]
> * The search icon appears broken

Noting this does not appear to be the case on the basic header: https://pr-37.d6umhtb6a6pvv.amplifyapp.com/components/detail/header--basic.html

**Review by dhcole — APPROVED** [review]

**Comment by dhcole** [comment]
The multiselect accordian doesn't appear to be multiselect: https://pr-37.d6umhtb6a6pvv.amplifyapp.com/components/preview/accordion--multiselectable.html

**Review by dhcole — CHANGES_REQUESTED** [review]
Just noticed... the USWDS repo changed its file structure, so we actually have a bunch of duplicate components now. See here: https://github.com/newjersey/njwds/tree/upgrade-uswds/src/components

Before, they had components in directories prefixed with numbers for sorting, but in a more recent change they removed these numbers, so many components appear with both formats.

Also noticing the most recent main branch in the USWDS repo looks like it's switched to storybook and done away with these components altogether...so I'm not actually sure where these new components are coming from, unless you have them locally from a previous (but more recent) uwds version. https://github.com/uswds/uswds/tree/v3.5.0/src

**Comment by dhcole** [comment]
Some discussion here... https://github.com/uswds/uswds/issues/3263 but not finding any clear communications on the rationale or migration strategy... I'd say we just clean up the duplication for now and then revisit what we want to be our upstream system as a more comprehensive task.

**Comment by namanaman** [comment]
@dhcole Thanks for looking into this further, I should have done some more careful inspection of these changes after importing... So it looks like `npm run import-components` is copying files from the `node_modules/uswds` directory. I realized we have a `node_modules/@uswds/uswds` directory, with the latest 3.5.0 NPM package, AND a `node_modules/uswds` directory with version 2.2.1, which is pulled as part of a `uswds-gulp` dependency. As you mentioned, the components I pulled in don't exist in USWDS 3.5.0, and have been replaced by Storybook stories in 3.0.0 and after. Based on that, I'm thinking the following:

- In this PR, I will upgrade to USWDS 3.5.0 but not import any new NJK components, to keep the behavior as it was. Anyway, these components are coming from an old package, so not much point in pulling them in.
- https://github.com/newjersey/njwds/issues/33 Use the new recommended `uswds-compile` instead of `uswds-gulp`. This will take away the confusing dependency of that extra older package in our system. This is useful anyway as `uswds-gulp` is being deprecated.
- https://github.com/newjersey/njwds/issues/38 Migrate Fractal to Storybook for NJWDS so we can easily import the USWDS stories upstream

File: src/components/card/card--flag.njk (hunk 4 above)
Comment (namanaman) [reply src/components/card/card--flag.njk:40]: Reverting back to not changing these. But we pull most in automatically from USWDS

**Review by dhcole — APPROVED** [review]
Looks great!

File: package.json [hunk 30]
```diff
@@ -23,7 +23,7 @@
   },
   "homepage": "https://github.com/newjersey/njwds#readme",
   "dependencies": {
-    "@uswds/uswds": "3.0.1"
+    "@uswds/uswds": "3.5.0"
```
Comment (dhcole) [inline package.json:26]: ✅  Confirming this is the latest release tag https://github.com/uswds/uswds/releases

**Pull request merged**